### ISSUE TITLE
Quotation border only printed on last page

### ISSUE THREAD

**Issue by meisterluk** [asciidoctor/asciidoctor-pdf#557]
**Given:**

```
Hello World
===========

[[someid]]
____
A


B


C


D


E


F


G


H


I


J


K


L


M


N


O


P


Q


R


S


T


U


V


W


X


Y


Z
____


Normal text afterwards.
```

**Expected:**
One border lefthandside to the quotation on both pages.

**Actual:**
First page shows no border. Second page has a border lefthandside to the quotation on both pages.
![quote-border](https://cloud.githubusercontent.com/assets/358358/18432595/1ce82500-78e3-11e6-9cf3-14b8d43d2b07.png)

**Comment by mojavelinux** [comment]
Framing for quote blocks that span multiple pages is not yet implemented. In fact, the only block type that currently supports this scenario (as of 1.5.0.alpha.13) are listing and literal blocks. The same treatment / logic needs to be applied to all blocks that have framing or background. This is a duplicate of #270.

Here are the issues to track multi-page support for various block types for reference:

* quote block - #270
* sidebar block - #259
* example block - #362

**Comment by meisterluk** [comment]
quote & verse is very different from implementing sidebar & example.

First call means lowest z-index. Drawing a background rectangle means it needs to be called before anything else. However, asciidoctor relies on prawnpdf to determine where to put characters or even page breaks. Hence in between setting text across multiple pages, it would be required to draw the rectangles.

Basically prawnpdf's API is too poorly engineered for such use cases. The only workaround I can see is manually interrupting setting text and checking whether `page_number` has changed. I won't fix it. I want nice solutions, not another workaround.

As far as this issue is concerned, a fix is available. I consider it fixed.

**Comment by mojavelinux** [comment]
We don't close issues until the PR is merged, so I'm going to reopen until then.

**Comment by mojavelinux** [comment]
I agree the Prawn API is poor at helping with layout, but I created a [two-phase writer](https://github.com/asciidoctor/asciidoctor-pdf/blob/a65b47409bc2e1410bea1e39b5121a9a45472f09/lib/asciidoctor-pdf/prawn_ext/extensions.rb#L802-L820) in the converter for precisely this use case. (I'll admit this API is still rough around the edges, but it gets the job done for now). The first phase performs a dry run that brings back the total height (it should also return the the start page and the end page, so I might add that). From that information, you can perform all the graphics work at the beginning of the second phase, then write the content.

**Comment by mojavelinux** [comment]
I'll share that it took me several weeks to come up with a solution for dealing with Prawn's single-pass rendering. That led me to using a scratch document, where I can test content before "burning" it into the main document.

**Comment by mojavelinux** [comment]
I've send a revised pull request to your pull request. If that looks good to you, merge it and I'll rebase and push upstream.

**Comment by mojavelinux** [comment]
Resolved by 471af3702929fccf89cd30149bf6399f9dec645f.

**Comment by meisterluk** [comment]
Thanks for improving it!
